### PR TITLE
Refactor: general cleanup from initial zustand PR

### PR DESCRIPTION
**Pull request by OwsleyJr**
#### Description

The objective of this PR was to cleanup a bit of the code from the previous Zustand PR as that one was meant to create our store structure mainly. So you will see a lot of folder changes/additions, code refactoring, and just better overall naming. I did notice a few spots where we are using ES5 so I decided to convert those to ES6, mainly where we are declaring our components. There is still a lot of work to be done but this will have improved our overall app structure and cleanliness.

The next PR will most likely be focused on creating a navbar to help us with navigation. We can most likely deprecate IndexScreen with this.

- Zustand global store has been renamed to useAppStore.
- Exercise list has been moved into it's own folder
- ES5 conversions to ES6.
- A new home page.
- New types folder

#### Screenshot (if UI-related)

N/A

#### To-Dos

- [x] Successful build `yarn ios/android`

### CODE REVIEW

File: packages/mobile/features/configure_workout/ConfigureWorkout.tsx [hunk 1]
```diff
@@ -35,7 +41,7 @@ function ConfigureWorkout({ navigation }: CreateWorkoutScreenProps) {
   };
 
   return (
-    <View>
+    <>
```
Comment (AndyG) [inline packages/mobile/features/configure_workout/ConfigureWorkout.tsx:44]: shouldn't need this fragment (i think)

File: packages/mobile/types/exercises.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export interface Exercise {
+  id: number;
+  name: string;
+}
+
+export interface ExercisesProps {
```
Comment (AndyG) [inline packages/mobile/types/exercises.ts:6]: is there benefit to having these here and not on the component itself?

File: packages/mobile/features/configure_workout/ConfigureWorkout.tsx (hunk 1 above)
Comment (OwsleyJr) [reply packages/mobile/features/configure_workout/ConfigureWorkout.tsx:44]: Needs a parent element.

File: packages/mobile/types/exercises.ts (hunk 2 above)
Comment (OwsleyJr) [reply packages/mobile/types/exercises.ts:6]: Probably don't need ExercisesProps now that I'm thinking about it. Can remove that and just keep Exercise

**Pull request merged**